### ISSUE TITLE
Semiprime factors ratio detection

### ISSUE THREAD

**Issue by sashakolpakov** [david-thrower/cerebros-core-algorithm-alpha#121]
This test is based on Sam Blake's preprint "Integer Factorisation, Fermat & Machine Learning on a Classical Computer", arXiv:2308.12290

Detecting the ratio of semiprime factors may theoretically help improving the classical Lawrence algorithm for semiprime factorization.

I used Cerebros on Sam Blake's data and got ~3% more false negatives. However, I gott ~10% better accuracy. Given that we used only 20% of the dataset for training, and 80% for testing, this result looks good (the dataset has 1e6 128-bit primes).

**Comment by david-thrower** [comment]
Paired with: Tabular binary classification in the Cerebros UI (We should be able to hyperparam tune this on the UI based system)

**Comment by david-thrower** [comment]
@sashakolpakov, One thought I have is that I just wonder if it is possible to quantize (or z / t / min-max scale) the series to coerce it to 32 bit precision.

I see a few issues that may affect the performance:

- By default, I think the weights are 32 bit signed floats. I may need to update the Cerebros API to include a kwarg for the data type of the weights, such that we can coerce the weights to match the datatype of the input data.
- A time series pattern may be identifiable from a quantized / scaled re-frame of the problem. I am curious if z / t / min-max scaling of the training data will improve the trade off in accuracy vs computational expense.
